utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -0,0 +1,1877 @@
+            // spatterColor.R = 5;
+            // spatterColor.G = 20;
+            // spatterColor.B = 64;
+
+            // Uncomment to run test case with an xywhROI override
+            /*for (i = 0; i < images; i++)
+            {
+                roiTensorPtrSrc[i].xywhROI.xy.x = 0;
+                roiTensorPtrSrc[i].xywhROI.xy.y = 0;
+                dstImgSizes[i].width = roiTensorPtrSrc[i].xywhROI.roiWidth = 100;
+                dstImgSizes[i].height = roiTensorPtrSrc[i].xywhROI.roiHeight = 180;
+            }*/
+
+            // Uncomment to run test case with an ltrbROI override
+            /*for (i = 0; i < images; i++)
+            {
+                roiTensorPtrSrc[i].ltrbROI.lt.x = 50;
+                roiTensorPtrSrc[i].ltrbROI.lt.y = 30;
+                roiTensorPtrSrc[i].ltrbROI.rb.x = 210;
+                roiTensorPtrSrc[i].ltrbROI.rb.y = 210;
+                dstImgSizes[i].width = roiTensorPtrSrc[i].ltrbROI.rb.x - roiTensorPtrSrc[i].ltrbROI.lt.x + 1;
+                dstImgSizes[i].height = roiTensorPtrSrc[i].ltrbROI.rb.y - roiTensorPtrSrc[i].ltrbROI.lt.y + 1;
+            }
+            roiTypeSrc = RpptRoiType::LTRB;
+            roiTypeDst = RpptRoiType::LTRB;*/
+
+            startOmp = omp_get_wtime();
+            start = clock();
+
+            if (inputBitDepth == 0 || inputBitDepth == 1 || inputBitDepth == 2 || inputBitDepth == 5)
+                rppt_spatter_gpu(d_input, srcDescPtr, d_output, dstDescPtr, spatterColor, roiTensorPtrSrc, roiTypeSrc, handle);
+            else
+                missingFuncFlag = 1;
+
+            break;
+        }
+        case 85:
+        {
+            testCaseName = "swap_channels";
+
+            startOmp = omp_get_wtime();
+            start = clock();
+
+            if (inputBitDepth == 0 || inputBitDepth == 1 || inputBitDepth == 2 || inputBitDepth == 5)
+                rppt_swap_channels_gpu(d_input, srcDescPtr, d_output, dstDescPtr, handle);
+            else
+                missingFuncFlag = 1;
+
+            break;
+        }
+        case 86:
+        {
+            testCaseName = "color_to_greyscale";
+
+            RpptSubpixelLayout srcSubpixelLayout = RpptSubpixelLayout::RGBtype;
+
+            startOmp = omp_get_wtime();
+            start = clock();
+            if (inputBitDepth == 0 || inputBitDepth == 1 || inputBitDepth == 2 || inputBitDepth == 5)
+                rppt_color_to_greyscale_gpu(d_input, srcDescPtr, d_output, dstDescPtr, srcSubpixelLayout, handle);
+            else
+                missingFuncFlag = 1;
+
+            break;
+        }
+        default:
+            missingFuncFlag = 1;
+            break;
+        }
+
+        hipDeviceSynchronize();
+        end = clock();
+        endOmp = omp_get_wtime();
+
+
+        if (missingFuncFlag == 1)
+        {
+            printf("\nThe functionality %s doesn't yet exist in RPP\n", func.c_str());
+            return -1;
+        }
+
+        gpuTime = ((double)(end - start)) / CLOCKS_PER_SEC;
```

Comment:
Lets change this to avoid any ambiguity henceforth.

HOST test suite:
- Measure startCpuTime, endCpuTime with clock() and startWallTime, endWallTime with omp_get_wtime()
- Compute cpuTime and wallTime
- Using std::max/min, and wallTime, find only maxWallTime, minWallTime and avgWallTime
- For display - call them CPU Backend Clock Time and CPU Backend Wall Time.
- Finally, I see below that the units are displayed as ms, but this line has / CLOCKS_PER_SEC that may give in seconds? Ensure all units in milliseconds.

HIP test suite:
- Please match everything with host test suite except that cpuTime using clock() doesn't make much sense, so lets remove it and only display "GPU Backend Wall Time = wallTime"
- Computation for avgWallTime remains same. Pls verify units.

Reply:
Done